projects/manager/src/app/providers/provider-editor/forms/aruna/aruna.component.ts

```diff
@@ -0,0 +1,127 @@
+import {Component, EventEmitter, Input, OnChanges, OnInit, Output, SimpleChanges} from '@angular/core';
+import {FormBuilder, FormControl, FormGroup, ReactiveFormsModule, Validators} from '@angular/forms';
+import {ArunaDataProviderDefinition, TypedDataProviderDefinition} from '@geoengine/openapi-client';
+import {isValidUuid} from '@geoengine/common';
+import {IdInputComponent} from '../util/id-input/id-input.component';
+import {MatCard, MatCardContent} from '@angular/material/card';
+import {MatFormField} from '@angular/material/form-field';
+import {MatInput, MatLabel} from '@angular/material/input';
+import {NgIf} from '@angular/common';
+import {CdkTextareaAutosize} from '@angular/cdk/text-field';
+import {MatTooltip} from '@angular/material/tooltip';
+
+@Component({
+    selector: 'geoengine-manager-aruna-editor-form',
+    templateUrl: './aruna.component.html',
+    styleUrl: './aruna.component.scss',
+    imports: [
+        IdInputComponent,
+        ReactiveFormsModule,
+        MatCard,
+        MatCardContent,
+        MatFormField,
+        MatLabel,
+        MatInput,
+        NgIf,
+        CdkTextareaAutosize,
+        MatTooltip,
+    ],
+})
+export class ArunaComponent implements OnInit, OnChanges {
+    @Input() provider?: TypedDataProviderDefinition;
+    @Input() createNew: boolean = false;
+    @Input() readonly: boolean = false;
+    @Input() visible: boolean = false;
+
+    @Output() changed = new EventEmitter<TypedDataProviderDefinition>();
+
+    form: FormGroup = this.fb.group({});
```

Comment:
can you make this a typed form group?

Reply:
https://github.com/geo-engine/geoengine-ui/pull/785/commits/885b207e0dea470a1ac095f875964789e38ba429